### DISCUSSION TITLE
Why doesn't Maui provide a cross-platform MeasureText method?

### DISCUSSION THREAD

**Discussion by DanTravison** [dotnet/maui#15957] [Q&A]
Goal: Given a text string and font attributes, what is rectangle needed to display the text.

I've seen discussions of measuring text with Skia and drawing on a canvas but I cannot find anything that allows me to measure text using Maui coordinates.  

One use case I encounter is setting the absolute width of a grid column to ensure it doesn't resize as the contents change. I know the maximum text length as well as the font and attributes but I don't see anywhere I can use this information to calculate the appropriate value for the width.

Does such a method exist?

**Comment by xtuzy** [comment]
It have a simple method to measure text, see https://github.com/dotnet/maui/blob/main/src/Graphics/src/Graphics/IStringSizeService.cs

**Comment by DanTravison** [comment]
I found that early on but the only implementation I found  was for Skia.  Did I miss the Maui coordinate space implementation?

**Reply by xtuzy** [reply]
please search what you want in source code before you ask question, https://github.com/search?q=repo%3Adotnet%2Fmaui+IStringSizeService&type=code

**Reply by Xyncgas** [reply]
knowing the font size like you said, there is a standard you can follow as to what rectangle size can 100% contain the string

**Reply by Stedy59** [reply]
There's no issue with asking the question. We all can learn from that exercise... And we encourage that.

**Comment by DanTravison** [comment]
I have found, in practice, that the IStringSizeService does not return a valid Size.Width, it is always less than the actual width of the text in pixels.
I have a few uses cases for this where MeasureOverride needs an accurate width and end up having to 'fudge' the measurement to ensure the size doesn't clip the contents.

 I've seen a few discussions about this limitation but no usable solution.
